vkrunner/vr-script.c

```diff
@@ -1923,108 +1958,181 @@ process_token_replacements(struct load_state *data)
+                if (!process_token_replacements(data)) {
                         res = false;
                         break;
                 }
 
-                res = process_line(&data);
+                res = process_line(data);
 
-                data.line_num++;
+                data->line_num++;
         } while (res);
 
         if (res)
-                res = end_section(&data);
+                res = end_section(data);
+
+        return res;
+}
+
+static struct vr_script *
+load_script_end(struct load_state *data)
+{
+        struct vr_script *script = data->script;
+
+        assert(data && "|data| must be a valid pointer.");
+        assert(script && "|script| must be a valid pointer.");
 
-        data.script->commands = (struct vr_script_command *) data.commands.data;
-        data.script->n_commands = (data.commands.length /
+        script->commands = (struct vr_script_command *) data->commands.data;
+        script->n_commands = (data->commands.length /
                                    sizeof (struct vr_script_command));
-        data.script->pipeline_keys =
-                (struct vr_pipeline_key *) data.pipeline_keys.data;
-        data.script->n_pipeline_keys = (data.pipeline_keys.length /
+        script->pipeline_keys =
+                (struct vr_pipeline_key *) data->pipeline_keys.data;
+        script->n_pipeline_keys = (data->pipeline_keys.length /
                                         sizeof (struct vr_pipeline_key));
-        data.script->extensions =
-                (const char *const *) data.extensions.data;
-        data.script->indices = (uint16_t *) data.indices.data;
-        data.script->n_indices =
-                data.indices.length / sizeof (uint16_t);
-
-        data.script->buffers = (struct vr_script_buffer *) data.buffers.data;
-        data.script->n_buffers = (data.buffers.length /
+        script->extensions =
+                (const char *const *) data->extensions.data;
+        script->indices = (uint16_t *) data->indices.data;
+        script->n_indices =
+                data->indices.length / sizeof (uint16_t);
+
+        script->buffers = (struct vr_script_buffer *) data->buffers.data;
+        script->n_buffers = (data->buffers.length /
                                   sizeof (struct vr_script_buffer));
-        qsort(data.script->buffers,
-              data.script->n_buffers,
-              sizeof data.script->buffers[0],
+        qsort(script->buffers,
+              script->n_buffers,
+              sizeof script->buffers[0],
               compare_buffer_binding);
 
-        vr_buffer_destroy(&data.buffer);
-        vr_buffer_destroy(&data.line);
+        vr_buffer_destroy(&data->buffer);
+        vr_buffer_destroy(&data->line);
+        vr_free(data);
 
-        if (res) {
-                return data.script;
-        } else {
-                vr_script_free(data.script);
-                return NULL;
-        }
+        return script;
```

Comment:
I think this bit of code could be retained here seeing as it is exactly duplicated in the two callers anyway.

Reply:
The reason why I moved it to those two callers is that `res` is not accessible in this function.
Let me pass `res` to this function and retain the code.